File: backend/app/schemas/routing.py
from pydantic import BaseModel, Field
from typing import List, Optional

class RouteRequest(BaseModel):
    origin: str = Field(..., example="Guwahati")
    destination: str = Field(..., example="Kohima")
    alpha: float = Field(0.5, description="Hazard weight sensitivity parameter", ge=0.0, le=10.0)

class DetourStep(BaseModel):
    instruction: str
    distance_km: float
    estimated_time_mins: float
    risk_score: float
    segment_name: str

class SafeRouteResponse(BaseModel):
    origin: str
    destination: str
    total_distance_km: float
    average_risk: float
    status: str  # OPEN, CAUTION, BLOCKED
    waypoints: List[List[float]] = Field(..., description="Array of coordinates [latitude, longitude]")
    detour_steps: List[DetourStep]
    alternative_available: bool = False

File: frontend/src/types/routing.ts
export interface DetourStep {
  instruction: string;
  distance_km: number;
  estimated_time_mins: number;
  risk_score: number;
  segment_name: string;
}

export interface SafeRouteResponse {
  origin: string;
  destination: string;
  total_distance_km: number;
  average_risk: number;
  status?: string; // Optional since it might be omitted
  waypoints: [number, number][];
  blocked_waypoints?: [number, number][];
  detour_steps: DetourStep[];
  alternative_available: boolean;
  detour_difference?: {
    extra_km: number;
    extra_mins: number;
    safety_score: number;
    helpline: string;
  };
}
